### PR TITLE
Google Auth (for review)

### PR DESCRIPTION
**Pull request by Gagnar1999**
google auth but it has some issues please review it and let me know what can i do it to imporve.
(also if you use this to debug your apk google sign in will not work we have to register sha1 of each device to get it worked) [currently in this branch i am using my own firebase google-services.json]

### CODE REVIEW

File: app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthScreen.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    navController: NavController, viewModel: AuthViewModel
+) {
+
+    val status by viewModel.loadingState.collectAsState()
+    val userSignedIn by viewModel.userSignedIn.collectAsState(initial = false)
+    var isLoading by remember { mutableStateOf(false) }
+    var errorText by remember {
+        mutableStateOf("")
+    }
+    if(userSignedIn == true){
```
Comment (hadiyarajesh) [inline app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthScreen.kt:53]: `if(userSignedIn == true)` can be simplified as `if(userSignedIn)`

File: app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthScreen.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                    viewModel.signInWithGoogle(credential)
+                } catch (e: Exception) {
+                    Log.e("TAG", "Google Sign In Failed ${e.localizedMessage}")
+                }
+            }
+        }
+    AuthView(isLoading) {
+        val gso =
+            GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken("719734510327-fkl6rvtae915bkc6j62u0d987jge5d0u.apps.googleusercontent.com")
```
Comment (hadiyarajesh) [inline app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthScreen.kt:75]: define a constant in separate file (there should be one already created) and use the token value from it

File: app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthScreen.kt [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            VerticalSpacer(size = 40)
+            Image(
+                painter = painterResource(id = R.drawable.google),
+                contentDescription = "",
+                modifier = Modifier
+                    .padding(12.dp)
+                    .size(100.dp)
+            )
+            SignInButton(
+                text = "Sign in with Google",
```
Comment (hadiyarajesh) [inline app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthScreen.kt:113]: user `stringResource()` to fetch from string resource value

File: app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import javax.inject.Inject
+@HiltViewModel
+class AuthViewModel @Inject constructor(private val preferenceManager: PreferenceManager) : ViewModel() {
+    val loadingState = MutableStateFlow(LoadingState.IDLE)
```
Comment (hadiyarajesh) [inline app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthViewModel.kt:18]: Explose `StateFlow` to view, not`MutableStateFlow`. You can create a public/private pair to achieve that.

File: app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        scope = viewModelScope,
+        started = SharingStarted.Eagerly,
+        initialValue = false
+    )
+
+    fun signInWithGoogle(credential: AuthCredential) =viewModelScope.launch {
+        try{
+            loadingState.emit(LoadingState.LOADING)
+            Firebase.auth.signInWithCredential(credential).await()
+            saveUserSignedIn()
```
Comment (hadiyarajesh) [inline app/src/main/java/com/hadiyarajesh/easynotes/ui/auth/AuthViewModel.kt:29]: This should be done in repository. Also, you can use [flower](https://github.com/hadiyarajesh/flower) to make network request, save to db and monitor states (loading/success/errro)

**Review by hadiyarajesh — CHANGES_REQUESTED** [review]
Why are you using your own json file? Can we use the one provided in the project?

File: app/build.gradle.kts [hunk 6]
```diff
@@ -35,9 +35,9 @@ android {
 
     buildTypes {
         debug {
-            applicationIdSuffix = ".debug"
+           /* applicationIdSuffix = ".debug"*/
```
[the remark this answers is not in the document]
Comment (Gagnar1999) [reply app/build.gradle.kts:38]: It is not compiling when i am using it with google-services.json file

**Pull request closed without merging**